registry/client/errors.go

```diff
@@ -47,7 +47,11 @@ func parseHTTPErrorResponse(r io.Reader) error {
 	return errors
 }
 
-func handleErrorResponse(resp *http.Response) error {
+// HandleErrorResponse returns error parsed from HTTP response for an
+// unsuccessful HTTP response code (in the range 400 - 499 inclusive). An
+// UnexpectedHTTPStatusError returned for response code outside of expected
+// range.
+func HandleErrorResponse(resp *http.Response) error {
```

Comment:
Another utility function exposed in the client interface. I think it is fine for this PR but I would prefer to move these into a subpackage (`client/errors` or something) following this. The client package interface was intended to minimize cruft.